sos/cleaner/mappings/__init__.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2020 Red Hat, Inc. Jake Hunsaker <jhunsake@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import re
+
+from collections import OrderedDict
+from threading import Lock
+
+
+class SoSMap():
+    """Standardized way to store items with their obfuscated counterparts.
+
+    Each type of sanitization that SoSCleaner supports should have a
+    corresponding SoSMap() object, to allow for easy retrieval of obfuscated
+    items.
+    """
+
+    ignore_list = []
+
+    def __init__(self):
+        self.dataset = OrderedDict()
+        self.lock = Lock()
+
+    def ignore_item(self, item):
+        """Some items need to be completely ignored, for example link-local or
+        loopback addresses should not be obfuscated
+        """
+        for skip in self.ignore_matches:
+            if re.match(skip, item):
+                return True
+
+    def item_in_dataset_values(self, item):
+        if item in self.dataset.values():
+            return True
+        return False
```

Comment:
Just `return (item in self.dataset.values())` ?

Reply:
Ack